### PR TITLE
E2-1905: Adding ability to disable method swizzling for PN

### PR DESCRIPTION
**Pull request by milos1290**
What              | Where/Who
------------------|----------------------------------------
JIRA Issue        | [E2-1905](https://leanplum.atlassian.net/browse/E2-1905)
People Involved   | @milos1290 

[Notes on PR descriptions](https://leanplum.atlassian.net/wiki/spaces/PR/pages/288424408/Creating+a+GitHub+Pull+Request)

## Background
Allow users to manually register push notification token

## Implementation
Adds option to enable/disable (enabled by default) method swizzling of PN methods. When swizzling is disabled, user needs to manually call registration method to register new push token.

## Is this change backwards-compatible?
Yes

### CODE REVIEW

**Comment by e7mac** [comment]
lgtm. Were there any unit tests that we might be able to add for these?

File: Leanplum-SDK/Classes/Internal/LPConstants.m [hunk 1]
```diff
@@ -237,6 +238,7 @@ - (id)init {
 
 NSString *LP_PATH_DOCUMENTS = @"Leanplum_Resources";
 NSString *LP_PATH_BUNDLE = @"Leanplum_Bundle";
+NSString *LP_SWIZZLING_ENABLED = @"LeanplumSwizzlingEnabled";
```
Comment (e7mac) [inline Leanplum-SDK/Classes/Internal/LPConstants.m]: This will be read in from the plist and used to set the value of _enableMethodSwizzling, correct?

Comment (hrishileanplum) [reply Leanplum-SDK/Classes/Internal/LPConstants.m]: @milos1290 , yes i would like to understand the use case for enableMethodSwizzling

File: Leanplum-SDK/Classes/Internal/Leanplum.m [hunk 2]
```diff
@@ -1644,6 +1644,59 @@ + (void)handleNotification:(NSDictionary *)userInfo
     [[LPCountAggregator sharedAggregator] incrementCount:@"handle_notification"];
 }
```
Comment (hrishileanplum) [inline Leanplum-SDK/Classes/Internal/Leanplum.m]: Is this PR incomplete? I do not see the code where we check if swizzling should be enabled or disabled. Also, how does it help by moving all the code to leanplum.m?

File: Leanplum-SDK/Classes/Internal/LPConstants.m (hunk 1 above)
Comment (milos1290) [reply Leanplum-SDK/Classes/Internal/LPConstants.m]: @e7mac  _enableMethodSwizzling is not used anymore, i've removed it in latest PR.

File: Leanplum-SDK/Classes/Internal/Leanplum.m (hunk 2 above)
Comment (milos1290) [reply Leanplum-SDK/Classes/Internal/Leanplum.m]: Yes the PR is complete functionality wise. I still need to fix tests and add see if i can add test cases.

File: Example/Leanplum-SDK/Images.xcassets/AppIcon.appiconset/Contents.json [hunk 3]
```diff
@@ -84,10 +84,15 @@
       "idiom" : "ipad",
       "size" : "83.5x83.5",
       "scale" : "2x"
+    },
+    {
+      "idiom" : "ios-marketing",
```
Comment (hrishileanplum) [inline Example/Leanplum-SDK/Images.xcassets/AppIcon.appiconset/Contents.json]: Whats this image for?

File: Example/Leanplum-SDK/Images.xcassets/AppIcon.appiconset/Contents.json [hunk 4]
```diff
@@ -84,10 +84,15 @@
...
+      "size" : "1024x1024",
+      "scale" : "1x"
     }
   ],
   "info" : {
     "version" : 1,
     "author" : "xcode"
   }
-}
+}
```
Comment (e7mac) [inline Example/Leanplum-SDK/Images.xcassets/AppIcon.appiconset/Contents.json]: nit: add a newline pls?

File: Example/Leanplum-SDK/Images.xcassets/AppIcon.appiconset/Contents.json (hunk 3 above)
Comment (milos1290) [reply Example/Leanplum-SDK/Images.xcassets/AppIcon.appiconset/Contents.json]: no idea how that got committed, will remove it.

[the remark this answers is not in the document]
Comment (milos1290) [reply Example/Leanplum-SDK/Images.xcassets/AppIcon.appiconset/Contents.json]: no idea how that got committed, will remove it.

**Comment by milos1290** [comment]
> Can you add test cases please? Like test cases with Swizzling enabled/disabled? Will it call the respective methods?

Not sure if I can

**Pull request merged**